### PR TITLE
Touch-up CI to closer match Rustls CI

### PR DESCRIPTION
**Pull request by cpu**
I think this repo's CI configuration hasn't seen as much love as some of the other Rustls repos. This branch is a quick tune-up adding some things that we've found helpful in other repos. Best reviewed commit-by-commit.

After this lands I'll update the branch protection rules and enable the merge queue feature.

### CODE REVIEW

**Comment by cpu** [comment]
> rustls / msrv (ubuntu-20.04) (pull_request) In progress — This check has started...
> rustls / msrv (macos-latest) (pull_request) In progress — This check has started...
> rustls / msrv (windows-latest) (pull_request) In progress — This check has started...

Over in rustls/rustls we only check MSRV on Ubuntu 20.04. Should this repo do the same? It will save some CI time.

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -41,26 +41,26 @@ jobs:
           toolchain: ${{ matrix.rust }}
 
       - name: cargo build (debug; default features)
-        run: cargo build
+        run: cargo build --locked
```
Comment (djc) [inline .github/workflows/rust.yml:44]: I feel like this should be in the same commit as the one that added `Cargo.lock`?

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -85,6 +85,28 @@ jobs:
 
       - run: cargo check --locked --lib --all-features
 
+  minver:
```
Comment (djc) [inline .github/workflows/rust.yml:88]: I guess we use `minver` in other repositories but the name triggers more of an MSRV notion for me than a `minimal-versions` notion. Maybe expand it? IMO brevity doesn't buy us that much here.

**Review by djc — APPROVED** [review]

**Comment by ctz** [comment]
> Should this repo do the same? It will save some CI time.

I think the code being compiled in this repo differs quite a bit by platform; whereas in rustls there is ~zero platform-specific code.

**Comment by cpu** [comment]
> I think the code being compiled in this repo differs quite a bit by platform; whereas in rustls there is ~zero platform-specific code.

That makes sense, I'll leave it as-is.

**Pull request merged**